packages/databricks-vscode/src/configuration/auth/AuthProvider.ts

```diff
@@ -64,32 +64,62 @@ export abstract class AuthProvider {
      * This function should not throw an error and each implementing class must
      * handle it's own error messages and retry loops.
      */
-    protected abstract _check(): Promise<boolean>;
+    protected abstract _check(token?: CancellationToken): Promise<boolean>;
 
-    public async check(force = false, showProgress = true): Promise<boolean> {
+    public async check(
+        force = false,
+        showProgress = true,
+        cancellationToken?: CancellationToken
+    ): Promise<boolean> {
         if (force) {
             this.checked = false;
         }
         if (this.checked) {
             return true;
         }
 
-        const checkFn = async () => {
-            this.checked = await this._check();
+        const checkFn = async (token?: CancellationToken) => {
+            this.checked = await this._check(token);
         };
 
         if (!showProgress) {
-            await checkFn();
+            await checkFn(cancellationToken);
             return this.checked;
         }
 
+        let cancellationRequested = false;
+        let task: Promise<void> = Promise.resolve();
         await window.withProgress(
             {
                 location: ProgressLocation.Notification,
                 title: `Databricks: Logging in using ${this.describe()}`,
+                cancellable: true,
             },
-            async () => await checkFn()
+            async (progress, token) => {
+                task = checkFn(token);
+                await Promise.race([
+                    task,
+                    new Promise((resolve) =>
+                        token.onCancellationRequested(resolve)
+                    ),
+                ]);
+                cancellationRequested = token.isCancellationRequested;
+            }
         );
+        if (cancellationRequested) {
+            await window.withProgress(
+                {
+                    location: ProgressLocation.Notification,
+                    title: `Databricks: Cancelling login using ${this.describe()}`,
+                },
+                async () => {
+                    await task;
+                }
+            );
+            window.showErrorMessage("Databricks: Login cancelled");
```

Comment:
The progress will already show a notification, do we need another one here?

Reply:
Progress just disappears. We need a final notification showing that something has cancelled.